tests/ui/impl-trait/defined-by-trait-resolution.rs

```diff
@@ -0,0 +1,12 @@
+//! The trait query `foo: Fn() -> u8` is a valid defining use of RPIT.
```

Comment:
Could you add revision to this test and the other one demonstrating what it does in the new solver?

```
// revisions: current next
//[next] compile-flags: -Ztrait-solver=next
```

Reply:
(I expect it to pass in the new solver too)